tests/test_cli_sync.py

```diff
@@ -231,3 +231,16 @@ def test_sync_ask_accepted(check_call, runner):
     runner.invoke(cli, ["--ask", "--dry-run"], input="y\n")
 
     assert check_call.call_count == 2
+
+
+@mock.patch("piptools.sync.check_call")
+def test_sync_dry_run_returns_non_zero_exit_code(check_call, runner):
```

Comment:
```suggestion
def test_sync_dry_run_returns_non_zero_exit_code(runner):
```

I wonder, do we actually need to patch `check_call`?

Reply:
Oh, no, actually it is not required as the `sync` won't call `check_call` if ran in dry mode. My bad.